artifact-authors.md

```diff
@@ -0,0 +1,224 @@
+
+### Layer Content Format
+
+The content layer format is up to the artifact author and may utilize standard or custom formats. Authors are encouraged to utilize existing formats which may benefit from existing libraries to parse those formats. Authors may choose to persist individual files with standard or custom formats, such as `.json`, `.yaml`, `.bin`, `.sif`, or store a collection of files as a `.tar` archive. Optionally, the author MAY choose to compress the contents. The extension of the `layer.mediaType` MUST reflect the format of the blob and its optional compression.
+
+Artifacts authors may define different layers for a range of reasons:
+
+1. Split layers due to size:
+    - Splitting an artifact into multiple layers enables concurrent downloading.
+    - Some artifacts will benefit from common layers, as OCI Images have common base layers and/or common framework/runtime layers.  
+    In this case, the layers are ordered where each layer overlays the previous layer, creating a unified collection of files.  
+    All layer types are the same, with the position in the index being the important aspect.
+1. Split layers for different groupings, reuse and optimization:
+    - An artifact may have non-unique environmental config and runtime collections of files. The runtime layer is sent to a host, while the configuration information may be sent to another location on the same host, or to a different host altogether.
+    - Layers may split for different localization options, where an artifact client can determine which layers it requires for locales and language requirements. The runtime layer, or the localization layers may be individually updated, isolating the storage and download time for delta changes.
+    - Layers may include optional source code, licenses or software bill of materials (SBoM) that are required for legal requirements or policy management. Different elements of a pipeline may pull the layers they require, while saving bandwidth, memory and compute processing for the layers they don't require.
+    - In these cases, the ordered position of the layer may not be important, rather the sub-type of the layer identifies its purpose.
+
+#### De-duping Layers
+
+When designing layer formats, consider how a registry may de-dupe layers that share the same content across different artifacts. Layers may be shared across different instances of an artifact as they're persisted to a registry. The registry may handle de-duping based on the digest of each layer. Artifact layers types are not expected, nor required to be shared across different artifact types. A layer type that represents an OCI Image is not expected to be shared with a helm chart. However, two helm charts that have the same content may be de-duped within the registry. The `layer.mediaType` provides the definition of the contents.
```

Comment:
"A layer type that represents an OCI Image is not expected to be shared with a helm chart"

Yes, but if they do reference the same digest, they *will* likely be de-duped, and we should make that clear

Reply:
You are the garbage collection god, and you keep us all on our toes to make sure we can do an awesome job at cleaning up remnants. :) 
For this, I'm not sure we need to worry about that level of details. Some registries do de-duping, some don't, in some circumstances, to give higher concurrency on content delivery. The point here just to let authors think about how they manage their layer designs. I don't know if an artifact author cares how we manage the back end de-duping as much as they care about efficient downloading. I'm mostly trying to get authors to keep common content in a separate layer so two artifacts that share common content don't each have to download the same content. 
Or, unlike above, "less info is more, so people actually read it".
I think we will have a lot of "articles" around this repo to help people with best practices, like your point above.